src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -92,6 +92,238 @@ public JsonArray(Buffer buf) {
     }
   }
 
+  /** Create a JsonArray containing zero values.
+   *
+   * @return an empty JsonArray.
+   */
+  public static JsonArray of() {
+    return new JsonArray();
+  }
+
+  /** Create a JsonArray containing one value.
+   *
+   * @param v1 the single value
+   * @return a JsonArray containing the specified value.
+   */
+  public static JsonArray of(Object v1) {
+    JsonArray arr = new JsonArray(new ArrayList<>(1));
+
+    arr.add(v1);
+
+    return arr;
+  }
+
+  /** Create a JsonArray containing two values.
+   *
+   * @param v1 the first value
+   * @param v2 the second value
+   * @return a JsonArray containing the specified values.
+   */
+  public static JsonArray of(Object v1, Object v2) {
+    JsonArray arr = new JsonArray(new ArrayList<>(2));
+
+    arr.add(v1);
+    arr.add(v2);
+
+    return arr;
+  }
+
+  /** Create a JsonArray containing three values.
+   *
+   * @param v1 the first value
+   * @param v2 the second value
+   * @param v3 the third value
+   * @return a JsonArray containing the specified values.
+   */
+  public static JsonArray of(Object v1, Object v2, Object v3) {
```

Comment:
I think for this we can use varrags, can't we ?

Reply:
yes, JsonArray does not need a `noisy API`, I changed it yesterday and just pushed 👀

Reply:
Another thing is for JsonObject maybe we could do this:
```
  protected JsonObject(String k1, Object v1, Object... more) {

    // implicit nullcheck of args
    if (more.length == 0) {
      map = new LinkedHashMap<>(1);
      put(k1, v1);
      return;
    }

    if ((more.length & 1) != 0) {
      throw new IllegalArgumentException("The more args length is odd");
    }

    map = new LinkedHashMap<>(1 + (more.length >> 2));
    put(k1, v1);
    for (int i = 0; i + 1 < more.length; i += 2) {
      put(more[i].toString(), more[i + 1]);
    }
  }

...

  public static JsonObject of(String k1, Object v1, String k2, Object v2) {
    return new JsonObject(k1, v1, k2, v2);
  }

  public static JsonObject of(String k1, Object v1, String k2, Object v2, String k3, Object v3) {
    return new JsonObject(k1, v1, k2, v2, k3, v3);
  }

... upto 15 or 20
```

Reply:
Please check @vietj 👀